server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -488,14 +486,7 @@ public Indexer(String indexName,
                 indexColumns.add(new IndexColumn(ref.column(), fieldType, indexInputs));
             }
         }
-        if (returnValues == null) {
-            this.returnValueInputs = null;
-        } else {
-            this.returnValueInputs = new ArrayList<>(returnValues.length);
-            for (Symbol returnValue : returnValues) {
-                this.returnValueInputs.add(ctxForRefs.add(returnValue));
-            }
-        }
+        this.returnValues = returnValues;
```

Comment:
This is now deferred - hence `returnValues` and `ctxForRefs` are fields of the Indexers, We need to replace refs and only then create new inputs.